internal/armadactl/queue.go

```diff
@@ -68,6 +81,44 @@ func (a *App) GetQueue(name string) error {
 	return nil
 }
 
+func (a *App) getAllQueuesAsAPIQueue(queueNames []string, labels []string, inverse bool) ([]*api.Queue, error) {
+	queueContainsAllLabelsAndInProvidedNames := func(q *api.Queue) bool {
+		containsAllLabels := slices.AllFunc(labels, func(label string) bool {
+			// If the label is a key, map the labels slice to only keys
+			labelsToCompare := q.Labels
+			if len(strings.Split(label, "=")) == 1 {
+				labelsToCompare = slices.Map(q.Labels, func(queueLabel string) string { return strings.Split(queueLabel, "=")[0] })
+			}
+
+			return slices.Contains(labelsToCompare, label)
+		})
+		inQueues := len(queueNames) == 0 || slices.Contains(queueNames, q.Name)
+		return inverse != (containsAllLabels && inQueues)
+	}
+	queuesToReturn, err := a.Params.QueueAPI.GetAll()
+	if err != nil {
+		return nil, errors.Errorf("[armadactl.getAllQueuesAsAPIQueue] error getting all queues: %s", err)
+	}
+
+	// Filter function seems expensive enough to justify parallel filter
```

Comment:
Is this correct?

I'd be surprised if running a small function on a few thousand objects really justified the complexity of ParallelFilter - but possibly you've tested it and know otherwise